tools/cli/commands/release.js

```diff
@@ -283,31 +284,43 @@ export async function getReleaseVersion( argv ) {
  * Bumps the correct number.
  *
  * @param {Array} version - the arguments passed
+ * @param {string} project - the project we're working with.
  * @returns {Array} the bumped version.
  */
-export async function getVersionBump( version ) {
+export async function getVersionBump( version, project ) {
 	version = version.split( '.' );
 
-	// If there was a point release, remove the last number before bumping.
-	if ( version[ 2 ] ) {
-		version.pop();
-	}
-
+	// If we we're bumping just a dev-release version, e.g. x.y-a.z
 	if ( version[ 0 ] === 'a' ) {
 		version[ 1 ] = parseInt( version[ 1 ] ) + 1;
 		return version.join( '.' );
 	}
 
-	if ( version[ 1 ] === '9' ) {
-		// e.g, if the current version is 10.9 and we want 11.0
-		version[ 0 ] = parseInt( version[ 0 ] ) + 1;
-		version[ 1 ] = '0';
-	} else {
-		// e.g, if the current version is 10.8 and we want 10.9
-		version[ 1 ] = parseInt( version[ 1 ] ) + 1;
+	const versioningType = await readComposerJson( project ).extra.changelogger;
```

Comment:
`versioningType` isn't really a correct name. `changeloggerConfig` maybe?

Even better would be to use optional chaining, as above, but again `esm` breaks.
```js
	const versioningType = await readComposerJson( project ).extra?.changelogger?.versioning;
```

Reply:
Good idea, updated the variable name 👍